cf/read_write/netcdf/netcdfread.py

```diff
@@ -412,26 +326,76 @@ def _customize_read_vars(self):
+                  `cfdm.NetCDFArray.__getitem__`.
 
-                .. versionadded:: 3.6.0
+        :Parmaeters:
+
+            ncvar: `str`
+                The netCDF variable name of the variable to be read.
 
         :Returns:
 
-            `netCDF4.Dataset`
-                The object for the file.
+            `numpy.ndarray`
+                The array containing the variable's data.
 
         """
-        out = super().file_open(filename, flatten=flatten, verbose=verbose)
-        _file_to_fh["netCDF"].pop(filename, None)
-        return out
+        variable = self.read_vars["variable_dataset"][ncvar][ncvar]
+        array = variable[...]
+
+        string_type = isinstance(array, str)
+        if string_type:
+            # --------------------------------------------------------
+            # A netCDF string type scalar variable comes out as Python
+            # str object, so convert it to a numpy array.
+            # --------------------------------------------------------
+            array = np.array(array, dtype=f"S{len(array)}")
+
+        if not variable.ndim:
+            # Hmm netCDF4 has a thing for making scalar size 1 , 1d
+            array = array.squeeze()
+
+        kind = array.dtype.kind
+        if not string_type and kind in "SU":
+            # --------------------------------------------------------
+            # Collapse (by concatenation) the outermost (fastest
+            # varying) dimension of char array into
+            # memory. E.g. [['a','b','c']] becomes ['abc']
+            # --------------------------------------------------------
+            if kind == "U":
+                array = array.astype("S")
+
+            array = netCDF4.chartostring(array)
+            shape = array.shape
+            array = np.array([x.rstrip() for x in array.flat], dtype="S")
+            array = np.reshape(array, shape)
+            array = np.ma.masked_where(array == b"", array)
+
+        elif not string_type and kind == "O":
+            # --------------------------------------------------------
+            # A netCDF string type N-d (N>=1) variable comes out as a
+            # numpy object array, so convert it to numpy string array.
+            # --------------------------------------------------------
+            array = array.astype("S")  # , copy=False)
+
+            # --------------------------------------------------------
+            # netCDF4 does not auto-mask VLEN variable, so do it here.
+            # --------------------------------------------------------
+            array = np.ma.where(array == b"", np.ma.masked, array)
+
+        elif not string_type and kind == "O":
+            # --------------------------------------------------------
+            # A netCDF string type N-d (N>=1) variable comes out as a
+            # numpy object array, so convert it to numpy string array.
+            # --------------------------------------------------------
```

Comment:
Not that it matters or we win any awards for it (though the good commenting is noted, as it often is), but there's a satisfying alignment across all of the comments here :smiley:

Reply:
On second look, partly because one comment is the same as another, but still...